Remove most mentions of Core license from docs

rmloveland opened the pull request:
Addresses DOC-10665

Summary of changes:

- Review and remove all mentions of Core license from docs, with the following exceptions (which will be later work):

  - Licensing FAQs page, which will be updated via DOC-11203

  - Mentions of Core Changefeeds, about which the method of removal from docs is TBD

rmloveland commented:
@kathancox can you please look at my (hopefully minimal) changes to the backup docs in this PR?  to make sure i'm not making a huge mess in your product area? 🙇

also as noted in the description i have completely left out core changefeeds while we work towards a decision there -  

@taroface for you this is a general review request pls 🫡   

ofc feel free to opine on the backup changes as well from a writerly POV

kathancox commented on src/current/v24.3/create-schedule-for-backup.md:
```diff
@@ -9,12 +9,6 @@ docs_area: reference.sql
 
 For more information about creating, managing, monitoring, and restoring from a scheduled backup, see [Manage a Backup Schedule]({% link {{ page.version.version }}/manage-a-backup-schedule.md %}).
```
There is also a reference to Enterprise licenses in the Parameters table under `FULL BACKUP crontab`.

kathancox replied:
"If you do not have an Enterprise license then you can only take full backups."

rmloveland replied:
Ah thanks! Updated to remove it

kathancox commented on src/current/v24.3/backup.md:
```diff
@@ -32,7 +32,7 @@ To view the contents of an backup created with the `BACKUP` statement, use [`SHO
 ## Considerations
 
-- Core users can only take [full backups]({% link {{ page.version.version }}/take-full-and-incremental-backups.md %}#full-backups). To use the other backup features, you need an [Enterprise license]({% link {{ page.version.version }}/enterprise-licensing.md %}). You can also use [CockroachDB {{ site.data.products.advanced }}](https://cockroachlabs.cloud/signup?referralId=docs-crdb-backup), which  ...
+If you use [CockroachDB {{ site.data.products.advanced }}](https://cockroachlabs.cloud/signup?referralId=docs-crdb-backup), it runs [full backups daily and incremental backups hourly]({% link cockroachcloud/managed-backups.md %}).
```
I think we should just remove this bullet point.

rmloveland replied:
Removed the whole thing! thanks

kathancox reviewed (commented):
Few comments, not sure if it is necessary to remove references of "This is an Enterprise-only feature".. if we do need to. Each of the Backup feature pages, like locality aware, encrypted etc. include a note saying this. Also, instances like this: https://www.cockroachlabs.com/docs/v24.2/backup-and-restore-overview#backup-and-restore-support (I'm not sure if the third column needs to change.) 

Let me know if there is anything I can do here to help. Also, I've sent a message out re Core changefeeds, will keep you up to date.

taroface commented on src/current/v24.3/sso-sql.md:
```diff
@@ -12,7 +12,7 @@ Cluster single sign-on (SSO) enables users to access the SQL interface of a Cock
 
 {{site.data.alerts.callout_info}}
-Cluster single sign-on for the DB Console is supported on {{ site.data.products.core }}, {{ site.data.products.enterprise }} and {{ site.data.products.advanced }} clusters. {{ site.data.products.standard }} and CockroachDB {{ site.data.products.basic }} clusters do not support cluster single sign-on. However, both CockroachDB {{ site.data.products.standard }} and CockroachDB {{ site.data.products ...
+Cluster single sign-on for the DB Console is supported on {{ site.data.products.core }} {{ site.data.products.enterprise }} and {{ site.data.products.advanced }} clusters. {{ site.data.products.standard }} and CockroachDB {{ site.data.products.basic }} clusters do not support cluster single sign-on. However, both CockroachDB {{ site.data.products.standard }} and CockroachDB {{ site.data.products. ...
```
This displays as: `Cluster single sign-on for the DB Console is supported on Self-Hosted Enterprise and Advanced clusters.`. However, the language is ambiguous because it could read as "self-hosted Enterprise and self-hosted Advanced clusters". It seems simplest to remove the `core` variable here so that it says `Cluster single sign-on for the DB Console is supported on Enterprise and Advanced clusters.` (especially since the `core` variable is currently misleading and should prob be renamed; I'll raise that to the team). 

Alternatively, we have it say `Cluster single sign-on for the DB Console is supported on self-hosted and Advanced clusters.` where `self-hosted` (lowercase) is displayed by a new product variable that means both Enterprise and Enterprise Free. To me, this is preferable but perhaps it's out of scope for this PR, which is about removing Core. I'll raise this separately also.

Links to the appropriate license/product pages would also help clarify this section, in case people need a refresher on what all these product names mean. Not sure what the best links would be?

```suggestion
Cluster single sign-on for the DB Console is supported on CockroachDB {{ site.data.products.enterprise }} and {{ site.data.products.advanced }} clusters. CockroachDB {{ site.data.products.standard }} and {{ site.data.products.basic }} clusters do not support cluster single sign-on. However, both CockroachDB {{ site.data.products.standard }} and CockroachDB {{ site.data.products.basic }} clusters can use [Cluster Single Sign-on (SSO) using `ccloud` and the CockroachDB {{ site.data.products.cloud }} Console]({% link cockroachcloud/cloud-sso-sql.md %}).
```

rmloveland replied:
Your suggested change appears to have been ~resolved by @mdlinville's changes in https://github.com/cockroachdb/docs/pull/19003 

It now reads 

> Cluster single sign-on for the DB Console is supported on {{ site.data.products.enterprise }} and {{ site.data.products.advanced }} clusters. {{ site.data.products.standard }} and CockroachDB {{ site.data.products.basic }} clusters do not support cluster single sign-on and do not have access to the DB Console. However, both CockroachDB {{ site.data.products.standard }} and CockroachDB {{ site.data.products.basic }} clusters can use [Cluster Single Sign-on (SSO) to authenticate to the `ccloud` command-line interface and to the CockroachDB Cloud Console]({% link cockroachcloud/cloud-sso-sql.md %})

I will rebase on `main` to pick this up

taroface commented on src/current/v24.3/ui-overview.md:
```diff
@@ -16,7 +16,7 @@ Refer to: [Network Authorization for CockroachDB Cloud Clusters&mdash;DB Console
 ## Authentication
 
-The DB Console supports username/password login, as well single sign-on (SSO) ({{ site.data.products.dedicated }} and {{ site.data.products.core }} {{ site.data.products.enterprise }} clusters only).
+The DB Console supports username/password login and single sign-on (SSO) for {{ site.data.products.dedicated }} and {{ site.data.products.core }} {{ site.data.products.enterprise }} clusters.
```
Same comment as earlier re: use of `core` variable. Also, `dedicated` needs to be replaced with presumably `advanced`? Someone else (@mdlinville ?) should verify.

```suggestion
The DB Console supports username/password login and single sign-on (SSO) for {{ site.data.products.dedicated }} and {{ site.data.products.enterprise }} clusters.
```

rmloveland replied:
I have applied your change to remove `core` as well as changed the `dedicated` to `advanced`!

taroface reviewed (changes requested):
Blocking here mostly for the Dedicated mention that slipped through. I have a longer comment on a separate change; sorry if you already have an upcoming plan for the things I raised! Otherwise I can raise it in Slack.

rmloveland commented:
> Few comments, not sure if it is necessary to remove references of "This is an Enterprise-only feature".. if we do need to. Each of the Backup feature pages, like locality aware, encrypted etc. include a note saying this. Also, instances like this: https://www.cockroachlabs.com/docs/v24.2/backup-and-restore-overview#backup-and-restore-support (I'm not sure if the third column needs to change.)

I have upscoped this PR to remove (almost) all such mentions of enterprise-only this and that.  Basically 99% of things that link to `enterprise-licensing.md` are now removed - the few that remain will be removed when I complete https://github.com/cockroachdb/docs/pull/19002 which will follow this

there's a lot of changed files but the changes are mostly trivial deletions of sentences like "This is an Enterprise feature [LINK]"  etc.

> And, instances like this "You do not need an [Enterprise license](https://www.cockroachlabs.com/docs/v24.2/enterprise-licensing) to run any of the backup validation tools described on this page."

Removed all of those as well! 

> Let me know if there is anything I can do here to help. Also, I've sent a message out re Core changefeeds, will keep you up to date.

awesome thanks! PTAL at the latest commit called `Update with kathancox feedback (1)`  which includes updates based on all of your feedback

rmloveland commented on src/current/v24.3/security-reference/security-overview.md:
```diff
@@ -20,15 +20,11 @@ Learn more: [Integrate CockroachDB {{ site.data.products.advanced }} with Satori
 CockroachDB {{ site.data.products.core }} here refers to the situation of a user deploying and operating their own cluster.
 
-{{ site.data.products.enterprise }} refers to an ongoing paid license relationship with Cockroach Labs. This license unlocks advanced features (see below). In this situation the customer maintains full control over their data, compute, and network resources while benefiting from the expertise of the Cockroach Labs's {{ site.data.products.enterprise }} Support staff.
+{{ site.data.products.enterprise }} refers to an ongoing license relationship with Cockroach Labs. In this situation the customer maintains full control over their data, compute, and network resources while benefiting from the expertise of the Cockroach Labs's {{ site.data.products.enterprise }} Support staff.
```
✅

rmloveland commented:
> Blocking here mostly for the Dedicated mention that slipped through. I have a longer comment on a separate change; sorry if you already have an upcoming plan for the things I raised! Otherwise I can raise it in Slack.

Thanks @taroface ! I have addressed your specific changes in the commit `Update with taroface feedback (1)`  - PTAL!

Re: the longer comment on a separate change / stuff we discussed in Slack earlier this week, I would like to address all those things too but I think for this specific "core deprecation" work those improvements will have to happen separately

taroface commented on src/current/v24.3/sso-sql.md:
```diff
@@ -14,7 +14,7 @@ Cluster single sign-on (SSO) enables users to access the SQL interface of a Cock
 
 {{site.data.alerts.callout_info}}
-Cluster single sign-on for the DB Console is supported on {{ site.data.products.enterprise }} and {{ site.data.products.advanced }} clusters. {{ site.data.products.standard }} and CockroachDB {{ site.data.products.basic }} clusters do not support cluster single sign-on and do not have access to the DB Console. However, both CockroachDB {{ site.data.products.standard }} and CockroachDB {{ site.dat ...
+Cluster single sign-on for the DB Console is supported on [{{ site.data.products.enterprise }}]({% link {{ page.version.version }}/licensing-faqs.md %}#types-of-licenses) and {{ site.data.products.advanced }} clusters. {{ site.data.products.standard }} and CockroachDB {{ site.data.products.basic }} clusters do not support cluster single sign-on and do not have access to the DB Console. However, b ...
```
```suggestion
Cluster single sign-on for the DB Console is supported on CockroachDB [{{ site.data.products.enterprise }}]({% link {{ page.version.version }}/licensing-faqs.md %}#types-of-licenses) and {{ site.data.products.advanced }} clusters. CockroachDB {{ site.data.products.standard }} and {{ site.data.products.basic }} clusters do not support cluster single sign-on and do not have access to the DB Console. However, both CockroachDB {{ site.data.products.standard }} and CockroachDB {{ site.data.products.basic }} clusters can use [Cluster Single Sign-on (SSO) to authenticate to the `ccloud` command-line interface and to the CockroachDB {{ site.data.products.cloud }} Console]({% link cockroachcloud/cloud-sso-sql.md %}).
```

rmloveland replied:
✅  these changes are in latest commit `Update with taroface feedback (2)`

taroface reviewed (approved):
LGTM, just a minor suggestion!

rmloveland commented:
@jaiayu FYI this is the "remove mention of Core and Enterprise from docs" PR I mentioned in today's weekly FYI

it does not talk about the new licensing, rather, it's the "rip out ~all the Core / Enterprise mentions" kind of pre-work PR

mdlinville reviewed (approved):
Since I was mentioned, I took a look at the includes and pages for deployment (install, orchestration, `start` commands, etc) and security / authentication, and they look great to me. Thank you so much. (the build failure is a handful of newly-broken anchors)

rmloveland commented:
synced with @kathancox offline and she's good with this merging

i will now fix a couple broken anchors and merge
